contributingGuides/STYLE.md

```diff
@@ -10,13 +69,514 @@ We use Prettier to automatically style our code.
+
+### No inline prop types
+
+Do not define prop types inline for components that are exported.
+
+> Why? Prop types might [need to be exported from component files](#export-prop-types). If the component is only used inside a file or module and not exported, then inline prop types can be used.
+
+```ts
+// BAD
+export default function MyComponent({ foo, bar }: { foo: string, bar: number }){
+   // component implementation
+};
+
+// GOOD
+type MyComponentProps = { foo: string, bar: number };
+export default MyComponent({ foo, bar }: MyComponentProps){
+   // component implementation
+}
+```
+
+### Satisfies Operator
+
+Use the `satisfies` operator when you need to validate that the structure of an expression matches a specific type, without affecting the resulting type of the expression.
+
+> Why? TypeScript developers often want to ensure that an expression aligns with a certain type, but they also want to retain the most specific type possible for inference purposes. The `satisfies` operator assists in doing both.
+
+```ts
+// BAD
+const sizingStyles = {
+    w50: {
+        width: '50%',
+    },
+    mw100: {
+        maxWidth: '100%',
+    },
+} as const;
+
+// GOOD
+const sizingStyles = {
+    w50: {
+        width: '50%',
+    },
+    mw100: {
+        maxWidth: '100%',
+    },
+} satisfies Record<string, ViewStyle>;
+```
+
+### Type imports/exports
+
+Always use the `type` keyword when importing/exporting types
+
+> Why? In order to improve code clarity and consistency and reduce bundle size after typescript transpilation, we enforce the all type imports/exports to contain the `type` keyword. This way, TypeScript can automatically remove those imports from the transpiled JavaScript bundle 
+
+Imports:
+```ts
+// BAD
+import {SomeType} from './a'
+import someVariable from './a'
+
+import {someVariable, SomeOtherType} from './b'
+
+// GOOD
+import type {SomeType} from './a'
+import someVariable from './a'
+```
+
+ Exports:
+```ts
+// BAD
+export {SomeType}
+export someVariable
+// or 
+export {someVariable, SomeOtherType}
+
+// GOOD
+export type {SomeType}
+export someVariable
+```
+
+### Exception to Rules
+
+Most of the rules are enforced in ESLint or checked by TypeScript. If you think your particular situation warrants an exception, post the context in the `#expensify-open-source` Slack channel with your message prefixed with `TS EXCEPTION:`. The internal engineer assigned to the PR should be the one that approves each exception, however all discussion regarding granting exceptions should happen in the public channel instead of the GitHub PR page so that the TS migration team can access them easily.
+
+When an exception is granted, link the relevant Slack conversation in your PR. Suppress ESLint or TypeScript warnings/errors with comments if necessary.
+
+This rule will apply until the migration is done. After the migration, discussion on granting exception can happen inside the PR page and doesn't need take place in the Slack channel.
```

Comment:
I think we could remove this paragraph, as this is a dead rule that nobody applied during migration 🤷‍♂️